launch/launch/exit_handler.py

```diff
@@ -23,17 +23,34 @@ def __init__(self, launch_state, task_state):
         self.task_state = task_state
 
 
-def default_exit_handler(context, ignore_returncode=False):
+def default_exit_handler(
+        context,
+        ignore_returncode=False,
+        teardown_on_error_only=False,
+        trigger_teardown=True):
     """
     Trigger teardown of launch.
 
     Use the returncode of the task for the launch if the launch was not already tearing down.
 
     @param ignore_returncode: If True, then ignore process return code.
+    @param teardown_on_error_only: If True, trigger launch context teardown
+    on exit if the return code indicates an error. This is mutually not
+    relevant if ignore_returncode is True or trigger_teardown is False.
+    @param trigger_teardown: If True, trigger launch context teardown
+    on exit if not already triggered.
```

Comment:
I'd be happy to change the keywords. These were the ones that made sense to me. They did accumulate over time.

My vision of this is that there's a combinatoric set of possible exit handlers. And if each combo can be combined in a single implementation then a thin wrapper added on top which calls it with the right argument that will be easier to maintain. (Like ignore_signal_exit_handler vs ignore_signal_exit_handler_no_teardown)

Similarly the default signal handler provides arguments such that the behavior can be modified from higher level handlers, but we keep a single implementation for each behavior based on the arguments passed through. This makes it relatively clear, how the different behaviors compare to each other.